### PR TITLE
Query multiple tables

### PR DESCRIPTION
**Pull request by Khaled-AlKhateeb**
##Tasks Done ✔️ 

1. Created a table named owners with (id, full_name, age) columns.

2. Create a table named species with (id, name) columns.

3. Modified the animals table: 
  - Made sure that id is set as autoincremented PRIMARY KEY.
  - Removed column species.
  - Added column species_id which is a foreign key referencing species table.
  - Added column owner_id which is a foreign key referencing the owners table.

4. Modified the schema.sql file.

5. Inserted the following data into the owners table:
  - Sam Smith 34 years old.
  - Jennifer Orwell 19 years old.
  - Bob 45 years old.
  - Melody Pond 77 years old.
  - Dean Winchester 14 years old.
  - Jodie Whittaker 38 years old.

6. Inserted the following data into the species table:
  - Pokemon.
  - Digimon.

7. Modified the inserted animals so it includes the species_id value:
  - If the name ends in "mon" it will be Digimon.
  - All other animals are Pokemon.

8. Modified the inserted animals to include owner information (owner_id):
  - Sam Smith owns Agumon.
  - Jennifer Orwell owns Gabumon and Pikachu.
  - Bob owns Devimon and Plantmon.
  - Melody Pond owns Charmander, Squirtle, and Blossom.
  - Dean Winchester owns Angemon and Boarmon.

9. Modified the data.sql file.

10. Wrote queries (using JOIN) to answer the following questions:
- What animals belong to Melody Pond?
4-melody_pond

- List of all animals that are pokemon (their type is Pokemon).
5-all_pokemons

- List all owners and their animals, remember to include those that don't own any animal.
6-all_owners

- How many animals are there per species?
7-animal_per_species

- List all Digimon owned by Jennifer Orwell.
8-digimon_jennifer

- List all animals owned by Dean Winchester that haven't tried to escape.
9-Dean-noEscape

- Who owns the most animals?
10-most_animals

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -16,58 +16,112 @@ SELECT * FROM animals WHERE name = 'Gabumon';
...
+GROUP BY species.name;
+
+-- [X] List all Digimon owned by Jennifer Orwell.
+SELECT owners.full_name AS Owner, animals.name As Animal, species.name AS Species
+FROM owners
+INNER JOIN animals
+ON owners.id = animals.owners_id
+INNER JOIN species
+ON species.id = animals.species_id
+WHERE species.name = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [x] To correctly write this query, you are supposed to select  from the animals table, then joining the owners and species table respectively. You could consider writing it as below

```
SELECT * FROM animals
   LEFT JOIN owners
   ON animals.owner_id = owners.id
   LEFT JOIN species
   ON animals.species_id = species.id
   WHERE owners.full_name = 'Jennifer Orwell'
   AND species.name = 'Digimon';
```

**Note:** you do not have to explicitly write is as mine is.

File: queries.sql [hunk 2]
```diff
@@ -16,58 +16,112 @@ SELECT * FROM animals WHERE name = 'Gabumon';
...
+
+-- [X] List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT owners.full_name AS Owner, animals.name As Animal, animals.escape_attempts
+FROM owners
+INNER JOIN animals
+ON owners.id = animals.owners_id
+WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [x] To correctly write this query, you are supposed to select  from the animals table, then joining the owners table. Kindly consider making this change.

File: queries.sql [hunk 3]
```diff
@@ -16,58 +16,112 @@ SELECT * FROM animals WHERE name = 'Gabumon';
...
+
+-- [X] Who owns the most animals?
+SELECT owners.full_name , COUNT(*)
+FROM owners
+INNER JOIN animals
+ON owners.id = animals.owners_id
+GROUP BY owners.full_name
+ORDER BY count DESC LIMIT 1;
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [x] To correctly write this query, you are supposed to select  from the animals table, then joining the owners table. Kindly consider making this change.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @Khaled-AlKhateeb    :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — APPROVED** [review]
### Hi @Khaled-AlKhateeb, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**